### PR TITLE
Make 'repository.path' required for 'build' and 'crossbuild' commands

### LINKED ISSUE #10: Use default import path in crossbuild or abort
The detected repo path in `promu info` is correct. It's not used in crossbuild though.

`docker run --rm -t -v $PWD:/app prom/golang-builder:1.5.3-main -i  -p 'linux/amd64 ...`

Instead of passing in an empty variable, promu should at least abort with something like `unknown repo URL`, but even better use the default provided by `repoLocation()`.

### PR DESCRIPTION
**Pull request by sdurrheimer**
Fixes #10

### CODE REVIEW

File: cmd/promu.go [hunk 1]
```diff
@@ -156,3 +156,12 @@ func envOr(name, def string) string {
 	}
 	return s
 }
+
+func checkRequiredConfigurations(configVars []string) {
+	for _, configVar := range configVars {
+		if !viper.IsSet(configVar) {
+			err := fmt.Errorf("'%s'", configVar)
+			fatalMsg(err, "Missing required configuration")
```
Comment (grobie) [inline cmd/promu.go]: The standard logging and fmt functions in go all have the interface `func(msg string, err error)`. Any reason to switch the arguments here?

**Comment by grobie** [comment]
:+1: I haven't tested it yet.

File: cmd/promu.go (hunk 1 above)
Comment (sdurrheimer) [inline cmd/promu.go]: Nope no reason. `fatalMsg(err error, msg string)` was a enhanced version of `fatal(err error)`.
I can change it in order to remain idiomatic to Go.

Moreover, I should have used `fatal()` with the full message as a `Error`.

Comment (sdurrheimer) [inline cmd/promu.go]: I've change the verification function to return a `bool` and a `error`. The `fatal()` is handled outside of the function.

File: cmd/promu.go [hunk 2]
```diff
@@ -165,3 +165,12 @@ func stringInSlice(needle string, haystack []string) bool {
 	}
 	return false
 }
+
+func hasRequiredConfigurations(configVars []string) (bool, error) {
```
Comment (grobie) [inline cmd/promu.go]: The return values seem to be dependent on each other. What about only returning `error` here. Either it's nil and there is no error or it returns the error message.

```go
if err := hasRequiredConfigurations([]string{"repository.path"}); err != nil {
    fatal(err)
}
```

**Comment by grobie** [comment]
What is `repoLocation()` used for? Can we remove it and read the value from .promu.yml instead?

File: cmd/promu.go (hunk 2 above)
Comment (sdurrheimer) [inline cmd/promu.go]: the `fatal()` function already check if `err != nil`.

Comment (grobie) [inline cmd/promu.go]: Ok. It's more common to leave that to the caller. `fatal(hasRequiredConfigurations(...))` adds a mental step.

**Comment by sdurrheimer** [comment]
Like I said in #10, current `repoLocation()` is useful for the `prom release` command. In case of a fork, you want to upload to your fork repository, not to the upstream repository.

**Comment by grobie** [comment]
Now I understand. It might make sense to move that into the release.go file to make the relationship clear, and display the promu.yml value in promu info (or show both).

**Comment by sdurrheimer** [comment]
> Now I understand. It might make sense to move that into the release.go file to make the relationship clear, and display the promu.yml value in promu info (or show both).

Not really, `repoLocation()` is used to complete the `ProjectInfo` struct. `release.go` is the only command using the `ProjectInfo.Owner` property currently, but others `ProjectInfo` properties are used in other commands.
Moreover, `repository.path` is useful for the `Go` context, but probably not for other languages if we ever decide/fell the need to support another one.

**Comment by sdurrheimer** [comment]
I've added a little refactoring on error handling.

File: cmd/promu.go [hunk 3]
```diff
@@ -165,3 +163,12 @@ func stringInSlice(needle string, haystack []string) bool {
 	}
 	return false
 }
+
+func hasRequiredConfigurations(configVars []string) error {
```
Comment (grobie) [inline cmd/promu.go]: Is there actually the need to check for multiple config vars at once? We could keep it simple until the need is there.

Another option is to use [variadic function](https://gobyexample.com/variadic-functions) to simplify the caller code.

```
func hasRequiredConfigurations(configVars ...string) error {
  // ...
}

hasRequiredConfigurations("config1", "config2")
```

File: cmd/release.go [hunk 4]
```diff
@@ -50,7 +50,7 @@ func runRelease(tarballsLocation string) {
 	}
 
 	err := filepath.Walk(tarballsLocation, uploadTarball)
-	fatalMsg(err, "Failed to upload tarballs")
+	fatalMsg("Failed to upload tarballs", err)
```
Comment (grobie) [inline cmd/release.go]: We need to check for `nil` here as well, right?

**Comment by grobie** [comment]
Looks good. Go is a bit more verbose than other languages, which might feel strange in the beginning, but it helps a lot to quickly read and understand code.

File: cmd/release.go (hunk 4 above)
Comment (sdurrheimer) [inline cmd/release.go]: Oh, I missed this one.

File: cmd/promu.go (hunk 3 above)
Comment (sdurrheimer) [inline cmd/promu.go]: Yes, I kinda like the variadic function way. I will do the change.

Comment (sdurrheimer) [inline cmd/promu.go]: Fixed

File: cmd/release.go (hunk 4 above)
Comment (sdurrheimer) [inline cmd/release.go]: Fixed

**Pull request merged**